packages/core-types/src/SmartAccountTypes.ts

```diff
@@ -67,6 +67,7 @@ export type SmartAccountState = {
   isDeployed: boolean // chain specific
   entryPointAddress: string // chain specific?
   implementationAddress: string
+  factoryAddress: string
```

Comment:
The initializeContracts function now requires SmartAccountState instead of ISmartAccount. It's necessary to include the factoryAddress field in SmartAccountState, previously required in ISmartAccount. This change doesn't cause any issues because the factoryAddress is only set in the getAddress function, which has the needed information.